auth0_flutter/lib/auth0_flutter_web.dart

```diff
@@ -94,6 +94,45 @@ class Auth0Web {
           scopes: scopes ?? {},
           idTokenValidationConfig: IdTokenValidationConfig(maxAge: maxAge)));
 
+  /// Opens a popup with the /authorize URL using the parameters provided as
+  /// arguments.
+  ///
+  /// Random and secure state and nonce parameters will be auto-generated. If
+  /// the response is successful, results will be valid according to
+  /// their expiration times.
+  ///
+  /// **Note**: This method should be called from an event handler that was
+  /// triggered by user interaction, such as a button click. Otherwise the
+  /// popup will be blocked in most browsers.
+  ///
+  /// Additonal notes:
+  ///
+  /// * [audience] relates to the API Identifier you want to reference in your access tokens (see [API settings](https://auth0.com/docs/get-started/apis/api-settings)).
+  /// * [scopes] defaults to `openid profile email`. You can
+  /// override these scopes, but `openid` is always requested regardless of
+  /// this setting.
+  /// * If you want to log into a specific organization, provide the
+  /// [organizationId]. Provide [invitationUrl] if a user has been invited
+  /// to join an organization.
+  /// * To provide your own popup window, create it using the `window.open()`
```

Comment:
```suggestion
  /// * To provide your own popup window, create it using the `html.window.open()`
  /// method from the dart:html package.
```

Reply:
Would it still work here? Or should we not expose the `popupWindow` param?

Reply:
And if it does work, should we type the param as `WindowBase` instead of `dynamic`?

Reply:
Yes it works.

```dart
final popup = window.open('', '', 'width=400,height=800');
final creds = await auth0Web.loginWithPopup(popupWindow: popup);
```

It does however require that `dart:html` is imported. This is why I initially typed it as `dynamic` in the API, but I've also just noticed that the developer would also receive [a warning](https://dart-lang.github.io/linter/lints/avoid_web_libraries_in_flutter.html) in the package plugin that says web libraries should not be used.

So perhaps we _can't_ support it without asking the developer to do some wrangling, for which we should provide an example. I'll leave it to you to decide how we proceed here.

Reply:
I would say that supporting a custom popup window is sometimes necessary depending on the browser, I think Safari is particular finicky about the popup not being opened directly from user interaction and will block the default one created by the SDK.

Reply:
Thanks for the clarification, I can see the issue. I'd suggest leaving it as `dynamic`, so it doesn't pull the `dart:html` simply by importing the SDK, but still allowing the developer to provide a custom popup if necessary.
I'd also suggest to document this in the API docs for the method.